agent/agent.go

```diff
@@ -199,6 +225,34 @@ func (a *Agent) startCheckTicker() error {
 	}
 }
 
+// Start a Ticker loop that will iterate over the hardcoded list of Discovery backends
+// and execute them. The initial run will happen relatively quickly after Agent launch
+// subsequent runs are done with a 15 minute delay. The effectiveness of the discoveries
+// is reported back in the "discover_cluster" Service in consul under a TTL of 60 minutes
+func (a *Agent) startDiscoverTicker() error {
+	ticker := time.NewTicker(a.cfg.DiscoverInterval / 2)
+	// Repeat after 15 minutes
+	if a.cfg.DiscoverInterval < time.Minute {
+		a.cfg.DiscoverInterval = 15 * time.Minute
+	}
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			for i := 0; i < len(a.discovers); i++ {
+				_, err := a.discovers[i]()
+				if err != nil {
+					log.Println("Error while discovering: ", err)
+				}
+			}
+			a.consul.Agent().UpdateTTL("discover_cluster", "", consul.HealthPassing)
+		case <-a.ctx.Done():
+			close(a.wsResultChan)
```

Comment:
I think this is a leftover from the other ticker. We shouldn't close this channel here.

Reply:
removed